### PR TITLE
Implement up-front sector pledge collateral calculation and tracking.

### PR DESCRIPTION
**Pull request by anorth**
Pledge is computed when a sector is proven and fixed for the sector's lifetime. The computation depends on the total network power at the time, so must be stored with the sector metadata, and a per-miner total maintained by the power actor. Removing the placeholder for dynamic pledge collateral allows a complete implementation.

### CODE REVIEW

File: actors/builtin/miner/miner_actor.go [hunk 1]
```diff
@@ -541,8 +539,9 @@ func (a *Actor) OnDeferredCronEvent(rt Runtime, params *OnDeferredCronEventParam
 	}
 
 	if payload.EventType == CronEventType_Miner_TempFault {
+		// should roll the huge number of state transactions in this loop into ~1
```
Comment (whyrusleeping) [inline actors/builtin/miner/miner_actor.go]:
```suggestion
		// TODO: should roll the huge number of state transactions in this loop into ~1
```

Comment (whyrusleeping) [reply actors/builtin/miner/miner_actor.go]: (makes this easier to grep for)

File: actors/builtin/power/power_state.go [hunk 2]
```diff
@@ -71,13 +87,13 @@ func (st *State) minerNominalPowerMeetsConsensusMinimum(s adt.Store, minerPower
 	}
 
 	var minerSizes []abi.StoragePower
-	var pwr abi.StoragePower
-	if err := adt.AsMap(s, st.ClaimedPower).ForEach(&pwr, func(k string) error {
+	var claimed Claim
```
Comment (whyrusleeping) [inline actors/builtin/power/power_state.go]: this method isnt used at all, and if it were, i would be throwing red flags at it from 50 feet away. Iterating the miner set during any onchain operation is asking for pain.

Can we just delete this one now?

Comment (anorth) [reply actors/builtin/power/power_state.go]: As discussed, the conditions higher up in this function mean this iteration happens only very early on in the network life, when no miner yet meets the size threshold.

This method *should* be called somewhere, as part of validating a miner is eligible for EPost. Maybe that call should come from off-chain.

File: actors/builtin/miner/miner_actor.go (hunk 1 above)
Comment (anorth) [reply actors/builtin/miner/miner_actor.go]: I've been reserving TODO temporarily for things that we absoluty must do, but this one is fixed by #135 anyway.

**Pull request merged**